### PR TITLE
[Reviewer: Rob] Add nonce-count support

### PR DESCRIPTION
**Pull request by mirw**
Rob,

Please can you review this enhancement to support nonce-count in quaff?

Thanks,

Matt

### CODE REVIEW

File: lib/auth.rb [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
-    def Auth.gen_nonce auth_pairs, username, passwd, method, sip_uri, ha1=""
+    def Auth.gen_nonce auth_pairs, username, passwd, method, sip_uri, ha1="", qop="", nc=1, cnonce=""
       if ha1.empty?
         a1 = username + ":" + auth_pairs["realm"] + ":" + passwd
         ha1 = Digest::MD5::hexdigest(a1)
       end
       a2 = method + ":" + sip_uri
       ha2 = Digest::MD5::hexdigest(a2)
-      digest = Digest::MD5.hexdigest(ha1 + ":" + auth_pairs["nonce"] + ":" + ha2)
+      if !qop.empty?
```
Comment (rkday) [inline lib/auth.rb]: Does it make sense to flip this around, and check for "qop == 'auth'" instead?

**Comment by mirw** [comment]
Thanks!  I've addressed your code review comment.  OK to merge?

**Comment by rkday** [comment]
Yep.

**Comment by rkday** [comment]
(Now at https://rubygems.org/gems/quaff/versions/0.7.3)

**Pull request merged**